includes/class-wc-subscription.php

```diff
@@ -1295,18 +1295,29 @@ protected function set_last_order_date( $date_type, $date = null ) {
 	 * @param string $date_type 'date_created', 'trial_end', 'next_payment', 'last_order_date_created', 'end' or 'end_of_prepaid_term'
 	 */
 	public function get_date_to_display( $date_type = 'next_payment' ) {
+		$date_type     = wcs_normalise_date_type_key( $date_type, true );
+		$timestamp_gmt = $this->get_time( $date_type, 'gmt' );
 
-		$date_type = wcs_normalise_date_type_key( $date_type, true );
+		return $this->format_date_to_display( $timestamp_gmt, $date_type );
+	}
 
```

Comment:
This change introduces a new function `format_date_to_display()`. Prior to this to display a date we'd call `get_date_to_display( $date_type );` and it would need to fetch the timestamp. 

In cases where we already have the timestamp, this would result in another get_time() call. This new function simply refactors out the formatting function of the original `get_date_to_display()` so if we have the timestamp, we can call `format_date_to_display()` instead.